src/Sparrow/Collections/FastStack.cs

```diff
@@ -170,16 +197,13 @@ public bool TryPeek(int depth, out T result)
         public T Pop()
         {
             if (_size == 0)
-                goto Error;
+                throw new InvalidOperationException("The stack is empty.");
 
             _version++;
             T item = _array[--_size];
-            _array[_size] = default(T);
+            _array[_size] = default;
 
             return item;
-
-            Error:
-            return ThrowForEmptyStack();
```

Comment:
Do we have other usages of this method? Can we remove it or should be replaced by direct `throw` like here?

Reply:
I copy it from v7.0 we haven't touched much this class since it was introduced, but since there are a few things we will have to do on it either way... I better apply the changes here and then we will up-merge it to v7.0